pkg/volume/azure_dd/azure_common_windows.go

```diff
@@ -0,0 +1,110 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure_dd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"k8s.io/utils/exec"
+)
+
+func scsiHostRescan(io ioHandler) {
+	//empty implementation here, don't need to rescan SCSI in Windows
+}
+
+func findDiskByLun(lun int, iohandler ioHandler, exe exec.Interface) (string, error) {
+	cmd := `Get-Disk | Where-Object { $_.location.contains("LUN") } | select number, location`
+	ex := exec.New()
+	output, err := ex.Command("powershell", "/c", cmd).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("Get-Disk failed in findDiskByLun, error: %v, output: %q", err, string(output))
+	}
+
+	if len(string(output)) < 10 {
+		return "", fmt.Errorf("Get-Disk output is too short, output: %q", string(output))
+	}
+
+	reader := bufio.NewReader(strings.NewReader(string(output)))
```

Comment:
parsing unstructured data is too error prone.  Consider adding 'ConvertTo-Json' to powershell cmdline and do json Unmarshalling.

Reply:
good point, I have used ConvertTo-Json and parse json format now